src/content-steering-controller.js

```diff
@@ -109,7 +110,8 @@ export default class ContentSteeringController extends videojs.EventTarget {
       this.decodeDataUriManifest_(steeringUri.substring(steeringUri.indexOf(',') + 1));
       return;
     }
-    this.steeringManifest.reloadUri = resolveUrl(baseUrl, steeringUri);
+    // With DASH queryBeforeStart, we want to use the steeringUri as soon as possible for the request.
+    this.steeringManifest.reloadUri = this.queryBeforeStart ? steeringUri : resolveUrl(baseUrl, steeringUri);
     // pathwayId is HLS defaultServiceLocation is DASH
     this.defaultPathway = steeringTag.pathwayId || steeringTag.defaultServiceLocation;
     // currently only DASH supports the following properties on <ContentSteering> tags.
```

Comment:
Curious, do we need this `if (this.manifestType_ === 'DASH')` can we just read tag values and set defaults if values are not presented?